packages/fast-usdc-contract/src/exos/status-manager.ts

```diff
@@ -402,22 +397,23 @@ export const prepareStatusManager = (
         nfa: NobleAddress,
         amount: bigint,
       ): { txHash: EvmHash; status: PendingTxStatus } | undefined {
-        const key = makePendingTxKey(nfa, amount);
-        if (!pendingSettleTxs.has(key)) return undefined;
-        const pending = pendingSettleTxs.get(key);
-
-        if (pending.length === 0) {
+        if (!pendingSettleTxs.has(nfa)) return undefined;
+        const allPending = pendingSettleTxs.get(nfa);
+        const idx = allPending.findIndex(tx => tx.tx.amount === amount);
+        if (idx < 0) {
           return undefined;
         }
         // extract first item
```

Comment:
this is extracting the leading range now, yes?

Reply:
Yes, the changes to `dequeueStatus` status had a short shelf life in this PR. "extract first item" is stale